nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/csv/CSVUtils.java

```diff
@@ -274,7 +275,12 @@ private static CSVFormat buildCustomFormat(final PropertyContext context, final
         final Character quoteChar = getCharUnescaped(context, QUOTE_CHAR, variables);
         format = format.withQuote(quoteChar);
 
-        final Character escapeChar = getCharUnescaped(context, ESCAPE_CHAR, variables);
+        final Character escapeChar;
+        if (context.getProperty(CSVUtils.ESCAPE_CHAR).evaluateAttributeExpressions(variables).getValue().isEmpty()) {
+            escapeChar = null;
+        } else {
+            escapeChar = getCharUnescaped(context, ESCAPE_CHAR, variables);
+        }
```

Comment:
Would the below be easier to read?

```suggestion
        final Character escapeChar = context.getProperty(CSVUtils.ESCAPE_CHAR).evaluateAttributeExpressions(variables).getValue().isEmpty() ? null : getCharUnescaped(context, ESCAPE_CHAR, variables);
```